docs/replacer.md

```diff
@@ -0,0 +1,34 @@
+# Replacer
+
+Replacer addons can be used to replace QRC resources at execution time. The addon must include all the required files in its bundle.
+
+The JSON files are validated via [JSON Schema](https://json-schema.org/) files:
+
+https://github.com/mozilla-mobile/mozilla-vpn-client/tree/main/scripts/ci/jsonSchemas
+
+On this page, we want to describe how to write a replacer addon.
+
+### Considerations
+
+ * Replacing QML files can be hazardous! The application will break if the QMLs are broken, or the dependencies are unsatisfied.
```

Comment:
+1 On this. Given that we currently have no way of sandboxing QML - this can be a huge code execution entry point. 

Maybe as a followup we should consider investigating if we can add a sec-context similar to principals, based on the `Q_INVOKABLE`'S `sender()`- so that i.e: 
A component in ://nebula// is not able to Access the VPNSettings Object. But a component in ://vpn can.